### PR TITLE
Migrate to library go

### PR DESCRIPTION
**Pull request by rgolangh**
library-go has helpers for creating an operator and registering controllers. All CSI driver 
that will be invoked using CSO are encouraged to use that library:

 	github.com/openshift/library-go

### CODE REVIEW

**Comment by rgolangh** [comment]
Now verifying. At least all objects are created and the `manifests` and `assets` dir seems most of it is working.

/woof

File: Makefile [hunk 1]
```diff
@@ -16,21 +16,33 @@ IMAGE_REGISTRY?=registry.svc.ci.openshift.org
...
+	targets/openshift/images.mk \
+	targets/openshift/bindata.mk \
+	targets/openshift/codegen.mk \
 )
 
+# Codegen module needs setting these required variables
+PACKAGE_BASE := github.com/ovirt/csi-driver-operator
+CODEGEN_OUTPUT_PACKAGE :=$(PACKAGE_BASE)/pkg/generated
+CODEGEN_API_PACKAGE :=$(PACKAGE_BASE)/pkg/apis
+CODEGEN_GROUPS_VERSION :=operator:v1alpha1
```
Comment (bertinatto) [inline Makefile]: We no longer need codegen since we should use `clustercsidrivers.operator.openshift.io` now

File: pkg/operator/starter.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+func RunOperator(ctx context.Context, controllerConfig *controllercmd.ControllerContext) error {
+	// Create clientsets and informers
+	kubeClient := kubeclient.NewForConfigOrDie(rest.AddUserAgent(controllerConfig.KubeConfig, operatorName))
+	dynamicClient := dynamicclient.NewForConfigOrDie(rest.AddUserAgent(controllerConfig.KubeConfig, operatorName))
+	kubeInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(kubeClient, defaultNamespace, "")
+
+	// Create GenericOperatorclient. This is used by the library-go controllers created down below
+	gvr := v1alpha1.SchemeGroupVersion.WithResource("ovirtcsidrivers")
+	operatorClient, dynamicInformers, err := goc.NewClusterScopedOperatorClient(controllerConfig.KubeConfig, gvr)
```
Comment (bertinatto) [inline pkg/operator/starter.go]: Please use `NewClusterScopedOperatorClientWithConfigName` instead; it has has been merged into library-go recently.

File: pkg/operator/starter.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+)
+
+func RunOperator(ctx context.Context, controllerConfig *controllercmd.ControllerContext) error {
+	// Create clientsets and informers
+	kubeClient := kubeclient.NewForConfigOrDie(rest.AddUserAgent(controllerConfig.KubeConfig, operatorName))
+	dynamicClient := dynamicclient.NewForConfigOrDie(rest.AddUserAgent(controllerConfig.KubeConfig, operatorName))
+	kubeInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(kubeClient, defaultNamespace, "")
+
+	// Create GenericOperatorclient. This is used by the library-go controllers created down below
+	gvr := v1alpha1.SchemeGroupVersion.WithResource("ovirtcsidrivers")
```
Comment (bertinatto) [inline pkg/operator/starter.go]: Now we can use `gvr := operatorv1.SchemeGroupVersion.WithResource("clustercsidrivers")` instead.

**Review by bertinatto — COMMENTED** [review]
The next step would be switching to the new `clustercsidrivers` resource that has been merged into openshift/api recently.

I'm working on that myself (EBS operator), you can check it at: https://github.com/openshift/aws-ebs-csi-driver-operator/pull/65

File: pkg/operator/starter.go (hunk 2 above)
Comment (rgolangh) [reply pkg/operator/starter.go]: ack

File: Makefile (hunk 1 above)
Comment (rgolangh) [reply Makefile]: ack

File: pkg/operator/starter.go (hunk 3 above)
Comment (rgolangh) [reply pkg/operator/starter.go]: ack

**Review by bennyz — APPROVED** [review]

**Pull request merged**